src/Servers/Kestrel/Core/src/Internal/KestrelServerImpl.cs

```diff
@@ -302,14 +297,16 @@ private async Task RebindAsync()
                     return;
                 }
 
+                Debug.Assert(Options.ConfigurationLoader != null); // TODO - Can ConfigurationLoader be null here?
+
                 reloadToken = Options.ConfigurationLoader.Configuration.GetReloadToken();
                 var (endpointsToStop, endpointsToStart) = Options.ConfigurationLoader.Reload();
 
                 Trace.LogDebug("Config reload token fired. Checking for changes...");
 
                 if (endpointsToStop.Count > 0)
                 {
-                    var urlsToStop = endpointsToStop.Select(lo => lo.EndpointConfig.Url ?? "<unknown>");
+                    var urlsToStop = endpointsToStop.Select(lo => lo.EndpointConfig?.Url ?? "<unknown>");
```

Comment:
Nit: ListenOptions in endpointsToStop must have a non-null EndpointConfig since it comes from the KestrelConfigurationLoader.

```suggestion
                    var urlsToStop = endpointsToStop.Select(lo => lo.EndpointConfig!.Url ?? "<unknown>");
```

Reply:
Url can't be null because ConfigurationReader throws an error when it is null or empty.

Can all the "<unknown>" be removed from this file?

Reply:
> Can all the "<unknown>" be removed from this file?

I believe so. If EndpointConfig isn't null, [Url shouldn't be null either.](https://github.com/dotnet/aspnetcore/blob/47138f9061f08af22f5df345051ff88354839a78/src/Servers/Kestrel/Core/src/Internal/ConfigurationReader.cs#L101)